docs/blog/2019-03-18-releasing-new-schema-customization/index.md

```diff
@@ -0,0 +1,131 @@
+
+First, install the latest and greatest version of `gatsby`, like so:
+
+```shell
+npm install gatsby --save
+```
+
+Next, continue reading below to see if any of the great, new features we've enabled scratch a particular itch. We feel very confident they will 💜
+
+# Recap of schema customization
+
+Before this change, the Gatsby GraphQL schema was generated automatically from the data that the user added to Gatsby. While very convenient and easy to start, changes to the data could cause changes to the schema, which could cause breakage in unrelated locations. Those bugs were confusing and hard to debug. To alleviate this problem, we've added a schema customization API that lets you customize, fix, and enhance types in your Gatsby GraphQL schema.
+
+There are two new APIs, `createTypes` and `createResolvers`.
+
+## `createTypes`
+
+`createTypes` can be used to define, fix, or extend a Gatsby GraphQL type that represents some of your app's data. Think of it like an escape hatch to politely inform Gatsby of your data's shape.
+
+```js:title=gatsby-node.js
+exports.sourceNodes = ({ actions }) => {
+  const { createTypes } = actions
+  const typeDefs = `
+    type AuthorJson implements Node {
+      name: String
+      birthday: Date
+    }
+  `
+  createTypes(typeDefs)
+}
+```
+
+After adding this to your [gatsby-node](/docs/gatsby-project-structure/#files) file, the `AuthorJson` type will always have fields name and birthday, regardless of the automatically inferred data shape. The rest of the fields will still be inferred normally, allowing you to enjoy the default benefits of Gatsby schema inference.
+
+## `createResolvers`
+
+`createResolvers` allows doing additional customization after all schema processing has been finished. Thus it can be used to add fields to any types, including root types like `Query` and types from third party schemas.
+
+```js:title=gatsby-node.js
+createResolvers({
+  Query: {
+    allAuthorFullNames: {
+      type: `[String!]!`,
+      resolve(source, args, context, info) {
+        const authors = context.nodeModel.getAllNodes({
+          type: `AuthorJson`,
+        })
+        return authors.map(author => author.name)
+      },
+    },
+  },
+})
+```
+
+## The Type Builder API
+
+While `createTypes` accepts `graphql-js` types along with a [Schema Definition Language (SDL)](https://www.prisma.io/blog/graphql-sdl-schema-definition-language-6755bcb9ce51) string, we've also added an option to use `graphql-js` types so that users could create types with resolvers. However, `graphql-js` is somewhat verbose and it can be hard to refer to types that don't yet exist or don't exist in a current scope. Therefore, we decided to add another programmatic API that combines brevity of SDL with flexibility of `graphql-js`.
```

Comment:
```suggestion
While `createTypes` accepts GraphQL types as a [Schema Definition Language (SDL)](https://graphql.org/learn/schema/) string, we've also added an option to use `graphql-js` types so that users could create types with resolvers. However, `graphql-js` is somewhat verbose and it can be hard to refer to types that don't yet exist or don't exist in a current scope. Therefore, we decided to add another programmatic API that combines brevity of SDL with flexibility of `graphql-js`.
```